spec/support/command_helpers.rb

```diff
@@ -38,13 +38,18 @@ def dummy_test_org
     DUMMY_TEST_ORG
   end
 
+  def dummy_test_app_global_identifier
+    @@global_identifier ||= SecureRandom.hex(2) # rubocop:disable Style/ClassVars
+  end
+
   # `extra_prefix` is used to differentiate between different dummy apps,
   # e.g., "dummy-test-default", "dummy-test-with-nothing", etc.
   #
   # Returns the full prefix.
   def dummy_test_app_prefix(extra_prefix = "")
     prefix = DUMMY_TEST_APP_PREFIX
     prefix += "-#{extra_prefix}" unless extra_prefix.nil? || extra_prefix.empty?
+    prefix += "-#{dummy_test_app_global_identifier}"
```

Comment:
We now add a global identifier at the end of the prefix, and since `@@global_identifier` is a class var, that ensures that the apps will have different names in different test runs.